packages/flutter/lib/src/widgets/binding.dart

```diff
@@ -93,8 +140,7 @@ class WidgetFlutterBinding extends FlutterBinding {
 
 /// Inflate the given widget and attach it to the screen.
 void runApp(Widget app) {
-  WidgetFlutterBinding.ensureInitialized();
-  WidgetFlutterBinding.instance._runApp(app);
+  new WidgetFlutterBinding()._runApp(app);
```

Comment:
Yeah the "caching" factory constructor is a bit weird. I can make that a static method instead with a more understandable name.